### PR TITLE
Enhance `find_outliers` and `identify_outliers` performance by avoiding duplication and filtering columns

### LINKED ISSUE #86: Avoid dataframe copies to reduce memory footprint of analysis functions
When working with large Pandas dataframes a large amount of memory is required to process the cosmicqc analysis functions, which utilize `df.copy()`. We could avoid this memory consumption by focusing on only the dataframe columns which are required, appending to existing dataframes when necessary (at a lower resource expense).

Relates to #23

### LINKED ISSUE #134: Read only relevant columns for processing to improve performance
`find_outliers` could see improved performance if it only read columns related to the outlier detection (avoiding a full read of the dataframe). Likely could overlap with improvements in #86 .

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR enhances `find_outliers` and `identify_outliers` performance by filtering to only the necessary columns and avoiding data duplication where possible. Along the path towards this PR I also analyzed code coverage and found we needed to clean up a bunch of legacy fixtures which are no longer used here (should boost the coverage percentage overall now as a result).

Closes #134 
Closes #86 

## What kind of change(s) are included?

- [ ] Documentation (changes docs or other related content)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [x] These changes pass all pre-commit checks.
- [x] I have added comments to my code to help provide understanding
- [ ] I have added a test which covers the code changes found within this PR
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: src/cosmicqc/analyze.py [hunk 1]
```diff
@@ -175,26 +178,29 @@ def find_outliers(
...
-            feature_thresholds=feature_thresholds,
-            feature_thresholds_file=feature_thresholds_file,
-        )
-    ]
+    # Determine the columns required for processing
+    required_columns = list(feature_thresholds.keys()) + metadata_columns
 
-    # Print outliers count and range for each feature
+    # Interpret the df as CytoDataFrame
+    df = CytoDataFrame(data=df)[required_columns]
```
Comment (jenna-tomkinson) [inline src/cosmicqc/analyze.py:192]: Is this a default behavior? I like the idea of improving speed here by not needing all the columns!

**Review by jenna-tomkinson — APPROVED** [review]
LGTM! 🎉  Love this small change! Just curious, is this behavior going to be the default and the user can choose to load in all columns?

File: src/cosmicqc/analyze.py (hunk 1 above)
Comment (d33bs) [reply src/cosmicqc/analyze.py:192]: This is a great question! I'll reassess here to ensure one can still read all the columns if necessary.

Comment (d33bs) [reply src/cosmicqc/analyze.py:192]: Opened #152

**Pull request merged**